python/ray/data/_internal/execution/operators/streaming_repartition_operator.py

```diff
@@ -0,0 +1,312 @@
+import logging
+from collections import deque
+from typing import Deque, Dict, Generator, List, Union
+
+import ray
+from ray.data._internal.delegating_block_builder import DelegatingBlockBuilder
+from ray.data._internal.execution.interfaces.execution_options import (
+    ExecutionOptions,
+    ExecutionResources,
+)
+from ray.data._internal.execution.interfaces.physical_operator import (
+    MetadataOpTask,
+    OpTask,
+    PhysicalOperator,
+)
+from ray.data._internal.execution.interfaces.ref_bundle import RefBundle
+from ray.data._internal.execution.operators.base_physical_operator import (
+    OneToOneOperator,
+)
+from ray.data._internal.stats import StatsDict
+from ray.data.block import Block, BlockAccessor, BlockMetadata
+from ray.data.context import DataContext
+from ray.types import ObjectRef
+
+logger = logging.getLogger(__name__)
+
+
+@ray.remote
+def _partition_block(
+    max_num_rows_per_block: int,
+    *input_blocks: Block,
+) -> Generator[Union[List[BlockMetadata], Block], None, None]:
+    """
+    Partition input blocks by maximum number of rows granularity using a generator
+    pattern.
+
+    Args:
+        max_num_rows_per_block: The maximum number of rows granularity for partition.
+        input_blocks: The input blocks to partition.
+
+    Yields:
+        - A list of all BlockMetadata as the first yield.
+        - Subsequent yields are individual blocks.
+    """
+
+    # Initialize lists to store metadata and partitioned blocks
+    metadata_list: List[BlockMetadata] = []
+    block_list: List[Block] = []
+
+    # Initialize a builder to accumulate rows into new blocks
+    cur_block_builder = DelegatingBlockBuilder()
+
+    # Iterate over each input block and partition based on max_num_rows_per_block
+    for block in input_blocks:
```

Comment:
I appreciate the thoroughness, but i don't think we should spare the comments for some of the logic that is complicated and requires a context and avoiding annotating trivial ones (like init, etc)